pywbemtools/pywbemcli/_cmd_connection.py

```diff
@@ -532,15 +532,16 @@ def cmd_connection_show(context, name, options):
     """
     Show the parameters that make up the current connection information if
     name is None. If Name exists, shows connections in connections file.
+    If name is "?" present list of all connections for selection.
```

Comment:
This docstring does not describe what the code does. The logic in the code is so complex that it is hard to describe what it actually does, and the comments in the code are also partly incorrect.

Can you try to get the docstring and the comments in sync with the code?

Also, I have to admit that after 10 minutes of trying to understand the logic in the code I gave up. This logic is really convoluted, or at least I cannot follow it this morning.

I think what would help is to explain the precedence of an inherited current connection in an interactive session, and a current connection defined in the connections file.